components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -173,135 +172,156 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	r.namedLogger().Debugw("received subscription reconciliation request", "namespace", req.Namespace, "name", req.Name)
 
-	actualSubscription := &eventingv1alpha1.Subscription{}
-	result := ctrl.Result{}
+	cachedSubscription := &eventingv1alpha1.Subscription{}
 
 	// Ensure the object was not deleted in the meantime
-	err := r.Client.Get(ctx, req.NamespacedName, actualSubscription)
+	err := r.Client.Get(ctx, req.NamespacedName, cachedSubscription)
 	if err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	// Handle only the new subscription
-	desiredSubscription := actualSubscription.DeepCopy()
+	subscription := cachedSubscription.DeepCopy()
 
 	// Bind fields to logger
-	log := utils.LoggerWithSubscription(r.namedLogger(), desiredSubscription)
+	log := utils.LoggerWithSubscription(r.namedLogger(), subscription)
 
-	if !desiredSubscription.ObjectMeta.DeletionTimestamp.IsZero() {
+	if !subscription.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
I think we are missing a return statement before this code block ends.

Reply:
Great! 
This is a very old issue introduced in January 2021, in this PR:
https://github.com/kyma-project/kyma/pull/10251/files#diff-34abfaea5dbe6be277421ffd2a2b5a7af2dfe19436d22ffab9bc8c3880dcc34bR94
This is why at the deletion of a subscription CR, in some cases, we had multiple reconcile() callbacks..

Reply:
done